app/controllers/CyaMaintainProfileController.scala

```diff
@@ -137,8 +135,44 @@ class CyaMaintainProfileController @Inject() (
             } yield Redirect(navigator.nextPage(CyaMaintainProfilePage, CheckMode, request.userAnswers))
           }
         case Left(errors)       =>
-          Future.successful(logErrorsAndContinue(errorMessage, continueUrl, errors))
+          Future.successful(logErrorsAndContinue(errorMessage, routes.ProfileController.onPageLoad(), errors))
       }
     }
 
+  def onPageLoadNirmsNumber(): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      traderProfileConnector.getTraderProfile(request.eori).flatMap { traderProfile =>
+        TraderProfile.buildNirms(request.userAnswers, request.eori, traderProfile) match {
+          case Right(_)     =>
+            val list = SummaryListViewModel(
+              rows = Seq(
+                HasNirmsSummary.rowUpdate(request.userAnswers),
+                NirmsNumberSummary.rowUpdate(request.userAnswers)
+              ).flatten
+            )
+            Future.successful(Ok(view(list, routes.CyaMaintainProfileController.onSubmitNirmsNumber)))
+          case Left(errors) =>
+            Future.successful(logErrorsAndContinue(errorMessage, routes.ProfileController.onPageLoad(), errors))
+        }
+      }
+  }
+
+  def onSubmitNirmsNumber(): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      traderProfileConnector.getTraderProfile(request.eori).flatMap { traderProfile =>
+        TraderProfile.buildNirms(request.userAnswers, request.eori, traderProfile) match {
+          case Right(model) =>
+            auditService.auditMaintainProfile(traderProfile, model, request.affinityGroup)
+            for {
+              _ <- traderProfileConnector.submitTraderProfile(model, request.eori)
```

Comment:
this doesn't need to be in a for comprehension because there is only one thing

Reply:
Done :)